packages/astro/src/runtime/server/astro-island.ts

```diff
@@ -57,8 +57,16 @@ declare const Astro: {
 				async childrenConnectedCallback() {
 					window.addEventListener('astro:hydrate', this.hydrate);
 					await import(this.getAttribute('before-hydration-url')!);
+					this.start();
+				}
+				start() {
 					const opts = JSON.parse(this.getAttribute('opts')!) as Record<string, any>;
-					Astro[this.getAttribute('client') as directiveAstroKeys](
+					const directive = this.getAttribute('client') as directiveAstroKeys;
+					if(Astro[directive] === undefined) {
+						window.addEventListener(`astro:${directive}`, () => this.start(), { once: true });
```

Comment:
nice and clean 👏

Reply:
()=>this.start() may be simply passed as this.start to make it cleaner